File: native-ui/app/src/main/java/com/example/cityinyourpocket/service/EventService.kt
package com.example.cityinyourpocket.service
import android.os.Build
import androidx.annotation.RequiresApi
import com.example.cityinyourpocket.model.Event
import com.example.cityinyourpocket.model.EventLocation
import kotlin.math.log

class EventService {
    private
    companion object{
        var  cities: MutableList<String> = mutableListOf()
        var events: MutableList<Event> = mutableListOf()
    }

    constructor() {
        if(cities.size == 0) {
            cities.add("Cluj-Napoca")
            cities.add("Bologa")
            cities.add("Bucharest")
            var loc1 = EventLocation(
                city = "Cluj-Napoca",
                latitude = 46.769896,
                longitude = 23.578607
            );
            var loc2 = EventLocation(
                city = "Bologa",
                latitude = 46.880920,
                longitude = 22.877560
            );
            var loc3 = EventLocation(
                city = "Bucharest",
                latitude = 44.470731,
                longitude = 26.082503
            )
            events.add(
                Event(
                    "Beer Festival", "Beer and more beer", "mirela",
                    loc1, "2020-10-20", "2020-10-22", "18:00", 1
                )
            )
            events.add(
                Event(
                    "Some History", "History and beer", "cristi",
                    loc2, "2020-10-20", "2020-10-22", "18:00", 2
                )
            )
            events.add(
                Event(
                    "Beer Festival", "Beer and more beer", "mirela",
                    loc3, "2020-10-20", "2020-10-22", "18:00", 3
                )
            )
            events.add(
                Event(
                    "Programming and pizza", "Lots of Java and some pizza", "cristi",
                    loc1, "2020-10-20", "2020-10-22", "18:00", 4
                )
            )
        }
    }

    fun cityHasEvents(city: String): Boolean{
        if(city=="all")
            return true
        for(currentCity in EventService.cities)
            if(currentCity == city)
                return true
        return false
    }


    fun getEventsFromCity(city: String): MutableList<Event>{
        if(city == "all"){
            var res = mutableListOf<Event>()
            for(event in EventService.events)
                res.add(event)
            return res
        }

        var result = mutableListOf<Event>()
        for(event in EventService.events)
            if(event.location.city == city)
                result.add(event);

        return  result;
    }

    fun getNextId(): Long{
        var nextID : Long = 0;
        for(event in events)
            if(nextID < event.evId)
                nextID = event.evId
        nextID+=1
        return nextID
    }

    fun addEvent(event: Event){
        val id : Long = getNextId()
        event.evId = id;
        events.add(event)
    }

    fun getUserOwnedEvents(user: String): MutableList<Event>{
        var result = mutableListOf<Event>()
        for(event in EventService.events)
            if(event.hostUserName == user)
                result.add(event);

        return  result;
    }


    fun removeEvent(id: Long): Boolean{
        var pos = -1;
        for(i in 0..events.size-1)
            if(events[i].evId == id)
                pos = i;
        if(pos>-1) {
            events.removeAt(pos)
            return true
        }
        return false
    }


    fun updateEvent(event: Event){
        for(ev in events){
            if(ev.evId == event.evId){
                ev.name = event.name
                ev.description = event.description
                ev.location.city = event.location.city
                ev.location.latitude = event.location.latitude
                ev.location.longitude = event.location.longitude
                ev.startDate = event.startDate
                ev.endDate = event.endDate
                ev.openHour = event.openHour
            }
        }
    }

    fun getEvent(id: Long): Event?{
        for(event in events)
            if(event.evId==id)
                return event
        return null
    }
}

File: non-native-ui/service/event.service.js
import {Event} from './../model/event';
import {EventLocation} from './../model/event-location';


export class EventService{
    static events = [
        new Event("Beer Festival",
        "Drinking beer online",
        "dan",
        new EventLocation("Cluj-Napoca",23.578607, 46.769896),
        "2020-11-11",
        "2020-11-12",
        "10:10",
        1
       ),
        new Event("Food mania",
        "lots of food and nice people",
        "dana",
        new EventLocation("Bucharest",26.082503,44.470731),
        "2020-11-12",
        "2020-11-12",
        "13:10",
        2),
        new Event("Titan Park reinvented",
        "paradise for kids and happiness for adults",
        "dana",
        new EventLocation("Bucharest",26.082503,44.470731),
        "2020-11-16",
        "2020-11-20",
        "15:10",
        3),
        new Event("Reading at the terrace",
        "Some good red wine, a good old book and a view over the city",
        "dan",
        new EventLocation("Cluj-Napoca",23.578607, 46.769896),
        "2020-11-11",
        "2020-11-12",
        "20:30",
        4),
        new Event("Programming and pizza",
        "Java, C# or something else? Not a problem, we are all united by the power of pizza",
        "alin",
        new EventLocation("Cluj-Napoca",23.578607, 46.769896),
        "2020-11-11",
        "2020-11-12",
        "20:30",
        5)
    ]


    constructor(){

    }


    eventsFromCity(city){
        result = []
        EventService.events.forEach(value => {
            if(value.location.city === city || city==="all")
            result.push(value)
        })
        return result
    }

    eventsOfUser(username){
        result = []
        EventService.events.forEach(value => {
            if(value.hostUserName === username)
            result.push(value)
        })
        return result
    }

    getEvent(eventId){
        result = null
        EventService.events.forEach(value =>{
            if(value.id === eventId){
                result =value
            }})
        return result
    }

    isOwnedByUser(username, evid){
        let event = this.getEvent(evid)
        if(event==null){
            return false
        }
        return event.hostUserName == username
    }

    removeEvent(evid){
        let index = -1;
        let i = 0;
        EventService.events.forEach(value =>{
            if(value.id === evid){
                index = i;
            }
            i += 1;
        })
        if (index > -1) {
          EventService.events.splice(index, 1);
        }
       
    }

    generateNextId(){
        let id = 1
        EventService.events.forEach(value => id+=1)
        return id
    }

    addEvent(name, description,username, startDate, endDate, startHour, location){
        let id = this.generateNextId()
        let event = new Event(name,
            description,
            username,
            new EventLocation(location, 0,0),
            startDate,
            endDate,
            startHour,
            id
        )
        EventService.events.push(event)
        return event
    }

    updateEvent(name, description,username, startDate, endDate, startHour, location,id){
            this.removeEvent(id);
            let event = new Event(name, description, username, new EventLocation(location,0,0), startDate, endDate, startHour, id);
            EventService.events.push(event);
            return event;
    }

}